wallet-new/test/unit/Test/Infrastructure/Generator.hs

```diff
@@ -124,8 +131,43 @@ cardanoModel boot = GeneratorModel {
     , gmMaxNumOurs    = 5
     }
 
-estimateCardanoFee :: Int -> Int -> Value
-estimateCardanoFee ins outs = fromIntegral $ (ins + outs) * 150000
+-- | Estimate the size of a transaction, in bytes.
+estimateSize :: Int      -- ^ Average size of @Attributes AddrAttributes@.
+             -> Int      -- ^ Size of transaction's @Attributes ()@.
+             -> Int      -- ^ Number of inputs to the transaction.
+             -> [Value]  -- ^ Coin value of each output to the transaction.
+             -> Byte     -- ^ Estimated size of the resulting transaction.
+estimateSize saa sta ins outs
+    = fromBytes . fromIntegral $
+      5
+    + 42 * ins
+    + (11 + listSize (32 + (fromIntegral saa))) * length outs
```

Comment:
I could do this, though I should warn that e.g. the 5 is actually a 2 + 2 + 1 with each term coming from a different part of the transaction. So it might end up a little messier.

The only values that are averages are the saa and sta parameters, plus 5 bytes per address for a CRC32 (it's actually encoded as variable length, but in practice it will almost always be 5 bytes)